automation_tools/satellite6/upgrade/tasks.py

```diff
@@ -504,34 +518,39 @@ def sync_tools_repos_to_upgrade(client_os, hosts):
         The AK name used in client subscription
     """
     client_os = client_os.upper()
-    tools_repo = os.environ.get('TOOLS_URL_{}'.format(client_os))
-    if tools_repo is None:
+    tools_repo_url = os.environ.get('TOOLS_URL_{}'.format(client_os))
+    if tools_repo_url is None:
         print('The Tools Repo URL for {} is not provided '
               'to perform Client Upgrade !'.format(client_os))
         sys.exit(1)
     cv_name, env_name, ak_name = [
         os.environ.get(env_var)
         for env_var in (
-            'RHEV_CLIENT_CV', 'RHEV_CLIENT_ENVIRONMENT', 'RHEV_CLIENT_AK')
+            'RHEV_CLIENT_CV_{}'.format(client_os),
+            'RHEV_CLIENT_ENVIRONMENT'.format(client_os),
+            'RHEV_CLIENT_AK_{}'.format(client_os)
+        )
     ]
-    details = os.environ.get('CLIENT_SUBSCRIPTION')
+    details = os.environ.get('CLIENT_SUBSCRIPTION_{}'.format(client_os))
     if details is not None:
         cv_name, env_name, ak_name = [
             item.strip() for item in details.split(',')]
     elif not all([cv_name, env_name, ak_name]):
-        print('Error! The CV, Env and AK details are not provided for Client'
-              'upgrade!')
+        print('Error! The CV, Env and AK details are not provided for {} '
+              'Client upgrade!'.format(client_os))
         sys.exit(1)
     # Set hammer configuration
     set_hammer_config()
-    hammer_product_create('tools6_latest', '1')
-    time.sleep(2)
-    hammer_repository_create(
-        'tools6_latest_repo', '1', 'tools6_latest', tools_repo)
-    hammer_repository_synchronize(
-        'tools6_latest_repo', '1', 'tools6_latest')
-    hammer_content_view_add_repository(
-        cv_name, '1', 'tools6_latest', 'tools6_latest_repo')
+    tools_product = 'tools6_latest_{}'.format(client_os)
+    tools_repo = 'tools6_latest_repo_{}'.format(client_os)
+    # adding sleeps in between to avoid race conditions
+    time.sleep(20)
```

Comment:
Actually there is bug logged in Bugzilla for race condition for creating product and repository one after another even the tasks are sync. I tried that but I am facing that race condition issue. So thought to put sleeps.